sql/parser/eval.go

```diff
@@ -259,6 +263,156 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// maxTime and minTime are not the theoretical max/min for the time.Time
```

Comment:
That is an excellent point, but we have to be careful here. `Time.Before` compares the `Time.sec` and `Time.nsec` fields, yet the behavior of `Time` for large values does not conform to those comparisons:

```
	t := &time.Time{}
	*t = t.Local()
	*(*int64)(unsafe.Pointer(t)) = math.MaxInt64
	fmt.Println(t)
	*(*int64)(unsafe.Pointer(t)) = -9223371966579706801
	fmt.Println(t)
```

This prints:

```
292277024627-12-06 10:30:07 -0500 EST
292277026854-11-09 07:00:15 -0500 EST
```

So the `Time.sec` field in the first line is greater than the `Time.sec` field in the second line, yet the date is less. I think we would have to use a definition like:

```
func (d DDate) IsMax() bool {
  n := d.Next().(DDate)
  return n.Unix() < d.Unix()
}
```

But I haven't convinced myself this is correct given the implementation of `Time.Before`.